junit/src/main/java/org/kaazing/k3po/junit/rules/Latch.java

```diff
@@ -28,6 +29,8 @@
     private final CountDownLatch startable;
     private final CountDownLatch finished;
     private volatile Thread testThread;
+    
+    private final AtomicBoolean exceptionOccurred = new AtomicBoolean(false); 
```

Comment:
Public API?  The latch code is our own...?  It can probably be (should be?) moved to internal package... Does any test use that API?? 

 I still don't get the setInterruptOnException call from the same Thread.   See JavaDoc on [interrupt()](https://docs.oracle.com/javase/7/docs/api/java/lang/Thread.html#interrupt()).  The only side effect of this call, when called from the current thread, is to set the interrupt flag (which could be transparently cleared depending on call stack / rules).  I don't see anywhere in the junit Statement javadoc or doc recommending this practice, and I can see how this could break other rules that might have a latch or block waiting for clean up (like what we would like to do with the Gateway Rule).

Happy to discuss the notify/await barrier, but my position is that as recommended in Code Complete, don't throw duplicate exceptions for a single cause.  Where ever the code went wrong that would cause a future await/notify barrier call to fail, (i.e. connection failure) we should be catching the exception there and displaying it properly.  If there is a case where await/notify barrier exceptions could be thrown and different then an existing exception, then it wouldn't be violating this best practice and I would change my opinion. I'd prefer doing the proper thing here in this PR, and then squashing newly visible bugs as they arise.